src/library-authoring/library-detail/LibraryAuthoringPage.jsx

```diff
@@ -0,0 +1,478 @@
+                      </h3>
+                    </Col>
+                    <Col xs={12} className="text-center py-3">
+                      <Button size="lg" block disabled={!hasChanges} onClick={commitChanges}>
+                        {intl.formatMessage(messages['library.detail.aside.publish'])}
+                      </Button>
+                    </Col>
+                    <Col xs={12} className="text-right">
+                      <Button variant="link" disabled={!hasChanges} onClick={revertChanges}>
+                        {intl.formatMessage(messages['library.detail.aside.discard'])}
+                      </Button>
+                    </Col>
+                  </Row>
+                </Card.Body>
+              </Card>
+            </Col>
+          </Row>
+        </aside>
+      </Col>
+    </Row>
+  </Container>
+);
+
+LibraryAuthoringPageBase.propTypes = {
+  intl: intlShape.isRequired,
+  library: libraryShape.isRequired,
+  getHandlerUrl: PropTypes.func.isRequired,
+  blockView: PropTypes.func.isRequired,
+  showPreviews: PropTypes.bool.isRequired,
+  setShowPreviews: PropTypes.func.isRequired,
+  sending: PropTypes.bool.isRequired,
+  newBlock: PropTypes.func.isRequired,
+  hasChanges: PropTypes.bool.isRequired,
+  revertChanges: PropTypes.func.isRequired,
+  commitChanges: PropTypes.func.isRequired,
+};
+
+const LibraryAuthoringPage = injectIntl(LibraryAuthoringPageBase);
+
+/**
+ * LibraryAuthoringPageContainerBase
+ *
+ * Container for the Library Authoring page.
+ * This is the main page for the authoring tool-- or it will be.
+ * It replaces LibraryPage. Currently, it only supports Video and Problem
+ * type libraries. Complex libraries will be supported in a later release.
+ */
+export const LibraryAuthoringPageContainerBase = ({
+  intl, library, blocks, ...props
+}) => {
+  // Explicit empty dependencies means on mount.
+  useEffect(() => {
+    const { libraryId } = props.match.params;
+    if (!library || (library && library.id !== libraryId)) {
+      props.clearLibrary().then(() => props.fetchLibraryDetail({ libraryId }));
+    }
+  }, []);
+  // If we end up needing this across components, or we end up needing more settings like this, we'll have to create
+  // another redux slice for 'common' settings which hydrates from localStorage.
+  let initialPreviewState = localStorage.getItem('showPreviews');
+  initialPreviewState = initialPreviewState ? JSON.parse(initialPreviewState) : true;
+  const [showPreviews, baseSetShowPreviews] = useState(initialPreviewState);
+  const setShowPreviews = (value) => {
+    localStorage.setItem('showPreviews', value);
+    baseSetShowPreviews(value);
+  };
+  const [sending, setSending] = useState(false);
+
+  const newBlock = () => {
+    setSending(true);
+    props.createLibraryBlock({
+      data: {
+        libraryId: library.id,
+        block_type: library.type,
+        definition_id: `${uuid4()}`,
+      },
+    }).finally(() => setSending(false));
```

Comment:
I think you also need to pass `libraryId` in the outer dictionary. As-is, this leads to `POST /api/libraries/v2/undefined/blocks` for me.